siliconcompiler/client.py

```diff
@@ -52,10 +52,11 @@ def remote_preprocess(chip):
     local_step = remote_steplist[0]
     indexlist = chip.getkeys('flowgraph', flow, local_step)
     for index in indexlist:
+        task = chip._get_task(local_step, index)
         tool = chip.get('flowgraph', flow, local_step, index, 'tool')
         # Setting up tool is optional (step may be a builtin function)
         if tool and tool not in chip.builtin:
-            chip._setup_tool(tool, local_step, index)
+            chip._setup_tool(tool, task, local_step, index)
```

Comment:
I suggest using: https://github.com/siliconcompiler/siliconcompiler/blob/b0d36a2a5f6013dd7c4a185837114c2436c09b6b/siliconcompiler/core.py#L4534
Otherwise this builtin check is not correct

Reply:
Hmm...this was written when a built-in `import` step seemed like a possibility, but I'm not sure if that is in the cards anymore.

Really, this pre-process method is intended to run `import` and then send the collected/linted inputs off for processing. Considering where we are now, it might make more sense to do something like error out if the task name is not `import`?

Reply:
you can use `_get_steps_by_task()['import']` to get the import steps (name of the step doesn't matter, but the task is import)

Reply:
Thanks, that's probably a good way to do it. I'm a little uneasy with the assumption that `import` tasks will always exist as the first N steps in a flow, but I guess I can add a check for that corner case.

Reply:
I don't see a problem using the new built-in check -- I think built-in import tasks are common (I've written lots of flows with a nop import), and the check is just here because builtins don't have a setup() method, right?

Adding an additional check that the first step is an "import" task seems fine, but not necessarily related to the builtin check?

Reply:
I'm just thinking through a more extensible remote workflow. It seems worthwhile to update the overall logic which finds and runs the `import` tasks, since I'm already fixing the syntax and we have more capable helper methods in the `Chip` object now.